File: proxy/1.java

import java.lang.reflect.*;
class MyDynamicProxy {
	public static void main(String [] args) {
		HelloImpl hello = new HelloImpl();
		MyInvocationHandler handler = new MyInvocationHandler(hello);
		Hello proxyHello = (Hello) Proxy.newProxyInstance(HelloImpl.class.getClassLoader(),HelloImpl.class.getInterfaces(),handler);
		proxyHello.sayHello();
	}

	interface Hello {
		void sayHello();
	}

	static class HelloImpl implements Hello {
		@Override
		public void sayHello() {
			System.out.println("Hello World");
		}
	}

	static class MyInvocationHandler implements InvocationHandler {
		private Object target;
		public MyInvocationHandler(Object target) {
			this.target = target;
		}

		@Override
		public Object invoke(Object proxy,Method method,Object[] args) throws Throwable {
			System.out.println("Invoking sayHello");
			Object result = method.invoke(target,args);
			return result;
		}
	}
}

File: cBase/c++/template/1.cpp
using namespace std;
#include <iostream>
#include <string>
template <typename T>
T add(T a,T b) {
    return a + b;
}

int main() {
    string s1 = "Hello";
    string s2 = " World";

    cout << add(12,13) << endl;
    cout << add(s1,s2) << endl;
    return 0;
}
